internal/resolution/variablesources/olm_variable_source.go

```diff
@@ -0,0 +1,81 @@
+package variablesources
+
+import (
+	"context"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/operator-framework/deppy/pkg/deppy"
+	rukpakv1alpha1 "github.com/operator-framework/rukpak/api/v1alpha1"
+
+	operatorsv1alpha1 "github.com/operator-framework/operator-controller/api/v1alpha1"
+	olmvariables "github.com/operator-framework/operator-controller/internal/resolution/variables"
+)
+
+type OLMVariableSource struct {
+	client        client.Client
+	catalogClient BundleProvider
+}
+
+func NewOLMVariableSource(cl client.Client, catalogClient BundleProvider) *OLMVariableSource {
+	return &OLMVariableSource{
+		client:        cl,
+		catalogClient: catalogClient,
+	}
+}
+
+func (o *OLMVariableSource) GetVariables(ctx context.Context) ([]deppy.Variable, error) {
+	operatorList := operatorsv1alpha1.OperatorList{}
+	if err := o.client.List(ctx, &operatorList); err != nil {
+		return nil, err
+	}
+
+	bundleDeploymentList := rukpakv1alpha1.BundleDeploymentList{}
+	if err := o.client.List(ctx, &bundleDeploymentList); err != nil {
+		return nil, err
+	}
+
+	allBundles, err := o.catalogClient.Bundles(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	requiredPackages := []*olmvariables.RequiredPackageVariable{}
+	for _, operator := range operatorList.Items {
+		requiredPackage, err := RequiredPackageVariable(allBundles, operator.Spec.PackageName, operator.Spec.Channel, operator.Spec.Version)
```

Comment:
The *Variable function names feel almost like constructors here, which normally would be fine - but the allBundles being passed everywhere gives you a sense that those variables aren't really carrying around "all the bundles" and there's actually some non-trivial amount of work being done under the covers. Maybe even like `Make*Variable`.

Reply:
or maybe something like: `Make*Variable(...).From(allBundles)`...?

Reply:
I added `Make` prefix. I'm not in favour of method chaining (at least not for at this point).